lib/matplotlib/image.py

```diff
@@ -420,13 +420,19 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                     a_min, a_max = np.int32(0), np.int32(1)
                 if inp_dtype.kind == 'f':
                     scaled_dtype = A.dtype
```

Comment:
perhaps the whole thing can be written as `scaled_dtype = np.float64 if A.dtype.itemsize > 4 else np.float32`?
Also, do we actually want to warn when upcasting float16 to float32 (which doesn't lose precision)?  For example we already silently cast ints to floats in many places in the lib.